cuqi/experimental/mcmc/_utilities.py

```diff
@@ -1,17 +1,164 @@
 import cuqi
 import inspect
+import numpy as np
 
-def find_valid_samplers(target):
-    """ Finds all samplers in the cuqi.experimental.mcmc module that accept the provided target. """
+# This import makes suggest_sampler easier to read
+import cuqi.experimental.mcmc as samplers 
+
+def find_valid_samplers(target, as_string = True):
+    """
+        Finds all possible sampleras that can be used for sampling from the target distribution.
+
+        Parameters
+        ----------
+
+        target : `cuqi.distribution.Distribution`
+            The target distribution to sample.
+        
+        as_string : boolean
+            Whether to return the name of the sampler as a string instead of instantiating a sampler. *Optional*
+
+    """
 
     all_samplers = [(name, cls) for name, cls in inspect.getmembers(cuqi.experimental.mcmc, inspect.isclass) if issubclass(cls, cuqi.experimental.mcmc.Sampler)]
     valid_samplers = []
 
     for name, sampler in all_samplers:
         try:
             sampler(target)
-            valid_samplers += [name]
+            valid_samplers += [name if as_string else sampler]
         except:
             pass
 
-    return valid_samplers
\ No newline at end of file
+    # Need a separate case for HybridGibbs
+    if find_valid_sampling_strategy(target) is not None:
+        valid_samplers += [cuqi.experimental.mcmc.HybridGibbs.__name__ if as_string else cuqi.experimental.mcmc.HybridGibbs]
+
+    return valid_samplers
+
+def find_valid_sampling_strategy(target, as_string = True):
```

Comment:
name suggestion: 
```suggestion
def find_valid_HybridGibbs_sampling_strategy(target, as_string = True):
```
for clarity